packages/grafana-ui/src/components/PieChart/PieChart.tsx

```diff
@@ -60,6 +60,21 @@ export const PieChart: FC<PieChartProps> = ({
   const theme = useTheme();
   const [state, setState] = useSetState<PieChartState>();
 
+  const setHighlightedSlice = (event: DataHoverEvent) => {
+    if (event.payload.source && eventBus?.isSourceOf(event.payload.source)) {
+      setState({ highlightedTitle: event.payload.dataId });
+    }
+  };
+
+  const resetHighlightedSlice = (event: DataHoverClearEvent) => {
+    if (event.payload.source && eventBus?.isSourceOf(event.payload.source)) {
+      setState({ highlightedTitle: undefined });
+    }
+  };
+
+  eventBus?.subscribe(DataHoverEvent, setHighlightedSlice);
+  eventBus?.subscribe(DataHoverClearEvent, resetHighlightedSlice);
```

Comment:
is it possible to wrap all these lines in if(eventBus) , would get rid of all the undefined ? checks